src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/scanneddocs/ManageScannedDocsContestedAboutToSubmitHandler.java

```diff
@@ -64,18 +68,27 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
         log.info("Received request to manage scanned documents for Case ID : {}", callbackRequest.getCaseDetails().getId());
         FinremCaseData caseData = callbackRequest.getCaseDetails().getData();
         List<String> warnings = new ArrayList<>();
-
         getValidatedResponse(caseData, warnings);
-        List<UploadCaseDocumentCollection> manageScannedDocumentCollection = caseData.getManageScannedDocumentCollection();
+
+        List<UploadCaseDocumentCollection> manageScannedDocumentCollection =
+            caseData.getManageScannedDocumentCollection().stream()
+                .filter(msdc -> YesOrNo.YES == msdc.getManageScannedDocument().getSelectForUpdate())
```

Comment:
better to use YesOrNo.isYes() static method

Reply:
Updated